pkg/grants/sync.go

```diff
@@ -19,7 +19,15 @@ func (g *Granter) SyncUser(log logr.Logger, namespace, rolePrefix string, user l
 	//   resolve required grants taking expiration into account
 	//   diff against existing
 	//   revoke/grant what is needed
-	accesses, err := g.groupAccesses(log, namespace, user.Spec.Read, user.Spec.Write)
+	read := []lunarwayv1alpha1.AccessSpec{}
+	if user.Spec.Read != nil {
+		read = *user.Spec.Read
+	}
+	write := []lunarwayv1alpha1.WriteAccessSpec{}
```

Comment:
These could be nil slices instead of empty slices until assigned a value?

```suggestion
	var read []lunarwayv1alpha1.AccessSpec
	if user.Spec.Read != nil {
		read = *user.Spec.Read
	}
	var write []lunarwayv1alpha1.WriteAccessSpec
```